build.gradle.kts

```diff
@@ -0,0 +1,663 @@
+    }
+
+    val versionValue = intellijVersions[versionKey] ?: return emptyList()
+    if (versionValue is List<*>) {
+        return versionValue as List<String>
+    }
+    return listOf(versionValue) as List<String>
+}
+
+tasks.register<UpdateEapBuildNumber>("updateEapBuildNumber")
+
+tasks.withType<DependencyUpdatesTask> {
+
+    val isStableVersion: (String) -> Boolean = { version ->
+        val stableKeyword = listOf("RELEASE", "FINAL", "GA").any { version.toUpperCase().contains(it) }
+        val regex = "/^[0-9,.v-]+(-r)?$/".toRegex()
+        stableKeyword || regex.matches(version)
+    }
+
+    rejectVersionIf {
+        !isStableVersion(this.candidate.version)
+    }
+}
+
+allprojects {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+    }
+
+    apply<JavaLibraryPlugin>()
+
+    java {
+        sourceCompatibility = javaMajorVersion
+        targetCompatibility = javaMajorVersion // redundant, added for clarity
+    }
+
+    // String interpolation support, see https://github.com/antkorwin/better-strings
+    // This needs to be enabled in each subproject by default because there's going to be no warning
+    // if this annotation processor isn't run in any subproject (the strings will be just interpreted verbatim, without interpolation applied).
+    // We'd only capture that in CI's post-compile checks by analyzing constants in class files.
+    dependencies {
+        annotationProcessor(rootProject.libs.betterStrings)
+        testAnnotationProcessor(rootProject.libs.betterStrings)
+    }
+
+    tasks.withType<JavaCompile> {
+        options.compilerArgs.addAll(listOf(
+                "-AcallToStringExplicitlyInInterpolations", // Enforce explicit `.toString()` call in code generated for string interpolations
+                "-Werror", // Treat each compiler warning (esp. the ones coming from Checker Framework) as an error.
+                "-Xlint:unchecked", // Warn of type-unsafe operations on generics.
+        ))
+
+        options.isFork = true
+        options.forkOptions.jvmArgs?.addAll(jvmArgsForJavaCompilation
+                ?: (if (isCI) listOf() else listOf("-Xmx6G", "-XX:+HeapDumpOnOutOfMemoryError")))
+
+        // `sourceCompatibility` and `targetCompatibility` say nothing about the Java APIs available to the compiled code.
+        // In fact, for X < Y it's perfectly possible to compile Java X code that uses Java Y APIs...
+        // This will work fine, until we actually try to run those compiled classes under Java X-compatible JVM,
+        // when we'll end up with NoSuchMethodError for APIs added between Java X and Java Y
+        // (i.e. for X=8 and Y=11: InputStream#readAllBytes, Stream#takeWhile and String#isBlank).
+        // `options.release = X` makes sure that regardless of Java version used to run the compiler,
+        // only Java X-compatible APIs are available to the compiled code.
+        // options.release = Integer.parseInt(javaMajorVersion.majorVersion)
```

Comment:
Whoops this shouldn't be commented out (pls search for other such hard-to-spot occurrences yourself as well)